### PR TITLE
Add Default Http Client

### PR DESCRIPTION
**Pull request by crobibero**
Adds the following headers by default:
| Header  | Value  |
|---|---|
| Accept-Encoding  | gzip, deflate  |
| User-Agent  | Jellyfin-Server/${Version} ~(${OS} team@jellyfin.org)~ |

User-Agent example: `Jellyfin-Server/10.7.0`

--
Removed string methods as compiler errors with due to https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2234?view=vs-2019

### CODE REVIEW

**Comment by ferferga** [comment]
Why the email address though?

**Comment by crobibero** [comment]
> 
> 
> Why the email address though?

I can remove it, it's currently only used for MusicBrainz, so I can remove it and have the MusicBrainz requests add it in

**Comment by ferferga** [comment]
@crobibero Oh, okay.

Then maybe it's better to keep it so we keep all the metadata providers hassle-free (maybe in the future more providers ask for such headers). 

At first, I thought you put that arbitrarily.

File: MediaBrowser.Common/Net/DefaultHttpClient.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MediaBrowser.Common.Net
+{
+    /// <summary>
+    /// Default http client.
+    /// </summary>
+    public class DefaultHttpClient
```
Comment (Bond-009) [inline MediaBrowser.Common/Net/DefaultHttpClient.cs:12]: Why is this needed?

Comment (crobibero) [reply MediaBrowser.Common/Net/DefaultHttpClient.cs:12]: Allows automatic handler usage, and I believe better practice.
Eventually we should move to specific HttpClients but this will allow the remaining IHttpClient removals

File: Jellyfin.Server/Startup.cs [hunk 2]
```diff
@@ -46,7 +52,20 @@ public void ConfigureServices(IServiceCollection services)
...
+                .AddHttpClient(NamedClient.Default, c =>
+                {
+                    c.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(_applicationHost.Name.Replace(' ', '-'), _applicationHost.ApplicationVersionString));
+                })
+                .ConfigurePrimaryHttpMessageHandler(x => new DefaultHttpClientHandler());
+
+            services.AddHttpClient(NamedClient.MusicBrainz, c =>
+                {
+                    c.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(_applicationHost.Name.Replace(' ', '-'), _applicationHost.ApplicationVersionString));
+                    c.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue($"({_applicationHost.ApplicationUserAgentAddress})"));
```
Comment (cvium) [inline Jellyfin.Server/Startup.cs:66]: https://docs.microsoft.com/en-us/dotnet/api/system.net.http.headers.productinfoheadervalue.-ctor?view=netcore-3.1#System_Net_Http_Headers_ProductInfoHeaderValue__ctor_System_String_

It's the ctor for adding a comment. Is that what you wanted?

Comment (cvium) [reply Jellyfin.Server/Startup.cs:66]: You can use ParseAdd instead of Add.

Comment (crobibero) [reply Jellyfin.Server/Startup.cs:66]: `The format of value 'team@jellyfin.org' is invalid.`
Looks like it would still need to be wrapped in parenthesis

File: Jellyfin.Server/Startup.cs [hunk 3]
```diff
@@ -46,7 +52,20 @@ public void ConfigureServices(IServiceCollection services)
             services.AddCustomAuthentication();
 
             services.AddJellyfinApiAuthorization();
-            services.AddHttpClient();
+
+            services
+                .AddHttpClient(NamedClient.Default, c =>
+                {
+                    c.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(_applicationHost.Name.Replace(' ', '-'), _applicationHost.ApplicationVersionString));
```
Comment (cvium) [inline Jellyfin.Server/Startup.cs:59]: Minor thing, but maybe put `new ProductInfoHeaderValue(_applicationHost.Name.Replace(' ', '-'), _applicationHost.ApplicationVersionString)` in a variable.

**Review by cvium — APPROVED** [review]

**Review by Ullmie02 — APPROVED** [review]
